Eq 14 add test setup

weapdiv-david opened the pull request:
**What**

Reimplements the healthtest and hello world page on new project layout. Modifies the test accordingly

**How to test**

_Pre-test_

a) Install the requirements for tests: `requirements_for_test.txt` 

1) Run the app using `scripts/run_app.sh`
2) Verify the Hello World page at the root of the web application using the browser
3) Click the healthcheck link and view the JSON
4) Run all the tests using `scripts/run_tests.sh`

**Who can test**

Anybody except @weapdiv-david

dhilton committed:
Added .gitignore file

dhilton committed:
Boiilerplate Flask app with automatic test discovery and templates

This commit adds a flask larger project layout; introducing blueprints, automatic test discovery,
standard test running script and project dynamic configuration.

We've also added holding error templates for 500, 404 and 503 http errors.

tafftoo committed:
Reimplemented healthcheck and modified tests

warrenbailey commented:
There's a lot more in this pull request than is described in the 'what' section. I think it's worth adding a sentence or two describing the initial flash application setup.

Also if you run the ./run_test.sh scripts the code style check fails as one of the python files has module imports that are not at the top of the file. I think we should fix that :-1:

dhilton commented:
Differing versions of pep8 - updating to the latest as we were using 1.5.7 and Warren was using 1.6.2 - we should use the latest and fix the issues raised.

dhilton committed:
Updated pep8 and added ignore for E402 issue

Version 1.6.2 introduced a new check for imports at the top of a file not having any other code lines before them.

This breaks a clear flask pattern of building Blueprint applications. Until that pattern is revised, we'll ignore that pep8 recommendation.

warrenbailey commented:
All tests now pass :+1:
